pkg/kubelet/kubelet.go

```diff
@@ -780,7 +780,11 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		containerRefManager,
 		kubeDeps.Recorder)
 
-	tokenManager := token.NewManager(kubeDeps.KubeClient.CoreV1())
+	var tokenManager *token.Manager
```

Comment:
Is https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/volume_host.go#L200 guaranteed not to be called in this case?

Otherwise, consider passing `kubeDeps.KubeClient` to `NewManager`, and handling the null case within `getToken(...)`

Reply:
Looks like the other managers aren't protected against this. Fixed anyway.